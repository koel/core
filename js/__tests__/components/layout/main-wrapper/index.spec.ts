import Component from '@/components/layout/main-wrapper/index.vue'
import { mount } from '@/__tests__/adapter'

describe('component/layout/main-wrapper/index', () => {
  it('renders properly', async () => {
    const wrapper = mount(Component)
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })
})
