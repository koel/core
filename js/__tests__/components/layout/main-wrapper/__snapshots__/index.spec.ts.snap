// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/layout/main-wrapper/index renders properly 1`] = `
<div id="mainWrapper">
  <nav id="sidebar" class="side side-nav showing">
    <section class="music">
      <h1>Your Music</h1>
      <ul class="menu">
        <li><a href="#!/home" class="home active">Home</a></li>
        <li><a href="#!/queue" class="queue">Current Queue</a></li>
        <li><a href="#!/songs" class="songs">All Songs</a></li>
        <li><a href="#!/albums" class="albums">Albums</a></li>
        <li><a href="#!/artists" class="artists">
            Artists
          </a></li>
        <!---->
      </ul>
    </section>
    <!---->
    <!---->
  </nav>
  <section id="mainContent">
    <!---->
    <!---->
    <section id="homeWrapper">
      <!---->
      <div class="main-scroll-wrap">
        <div class="two-cols">
          <!---->
          <section class="recent">
            <h1>
              Recently Played
              <!---->
            </h1>
            <!---->
            <p class="text-light-gray">
              Your recently played songs will be displayed here.<br>
              Start listening!
            </p>
          </section>
        </div>
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
    </section>
    <section id="queueWrapper" style="display: none;">
      <!---->
      <div tabindex="0" class="song-list-wrap main-scroll-wrap queue" style="display: none;">
        <table class="song-list-header sortable">
          <thead>
            <tr>
              <th class="track-number">
                #
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="title">
                Title
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="artist">
                Artist
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="album">
                Album
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="time">
                Time
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="favorite"></th>
              <th class="play"></th>
            </tr>
          </thead>
        </table>
        <div data-v-303cd025="" class="virtual-scroller scroller">
          <div data-v-303cd025="" class="item-container">
            <table data-v-303cd025="" class="items"></table>
          </div>
          <div data-v-c7b3205a="" data-v-303cd025="" tabindex="-1" class="resize-observer"><object style="display: block; position: absolute; top: 0; left: 0; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;" type="text/html" data="about:blank"></object></div>
        </div>
      </div>
      <div class="none text-light-gray">
        <p>Empty spaces. Abandoned places.</p>
        <!---->
      </div>
    </section>
    <section id="songsWrapper" style="display: none;">
      <!---->
      <div tabindex="0" class="song-list-wrap main-scroll-wrap all-songs">
        <table class="song-list-header sortable">
          <thead>
            <tr>
              <th class="track-number">
                #
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="title">
                Title
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="artist">
                Artist
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="album">
                Album
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="time">
                Time
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="favorite"></th>
              <th class="play"></th>
            </tr>
          </thead>
        </table>
        <div data-v-303cd025="" class="virtual-scroller scroller">
          <div data-v-303cd025="" class="item-container">
            <table data-v-303cd025="" class="items"></table>
          </div>
          <div data-v-c7b3205a="" data-v-303cd025="" tabindex="-1" class="resize-observer"><object style="display: block; position: absolute; top: 0; left: 0; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;" type="text/html" data="about:blank"></object></div>
        </div>
      </div>
    </section>
    <section id="albumsWrapper" style="display: none;">
      <!---->
      <div class="albums main-scroll-wrap as-">
        <!---->
      </div>
    </section>
    <section id="artistsWrapper" style="display: none;">
      <!---->
      <div class="artists main-scroll-wrap as-">
        <!---->
      </div>
    </section>
    <section id="playlistWrapper" style="display: none;">
      <!---->
      <!---->
    </section>
    <section id="favoritesWrapper" style="display: none;">
      <!---->
      <div tabindex="0" class="song-list-wrap main-scroll-wrap favorites" style="display: none;">
        <table class="song-list-header sortable">
          <thead>
            <tr>
              <th class="track-number">
                #
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="title">
                Title
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="artist">
                Artist
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="album">
                Album
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="time">
                Time
                <i class="fa fa-angle-down" style="display: none;"></i> <i class="fa fa-angle-up" style="display: none;"></i></th>
              <th class="favorite"></th>
              <th class="play"></th>
            </tr>
          </thead>
        </table>
        <div data-v-303cd025="" class="virtual-scroller scroller">
          <div data-v-303cd025="" class="item-container">
            <table data-v-303cd025="" class="items"></table>
          </div>
          <div data-v-c7b3205a="" data-v-303cd025="" tabindex="-1" class="resize-observer"><object style="display: block; position: absolute; top: 0; left: 0; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;" type="text/html" data="about:blank"></object></div>
        </div>
      </div>
      <div class="text-light-gray none">
        Start loving!
        Click the <i class="fa fa-heart" style="margin: 0px 5px;"></i> icon when a song is playing to add it
        to this list.
      </div>
    </section>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
  </section>
  <section id="extra" class="text-light-gray showing">
    <div class="tabs">
      <div role="tablist" class="clear"><button aria-selected="true" aria-controls="extraPanelLyrics" id="extraTabLyrics" role="tab">
          Lyrics
        </button> <button aria-controls="extraPanelArtist" id="extraTabArtist" role="tab">
          Artist
        </button> <button aria-controls="extraPanelAlbum" id="extraTabAlbum" role="tab">
          Album
        </button>
        <!---->
      </div>
      <div class="panes">
        <div aria-labelledby="extraTabLyrics" id="extraPanelLyrics" role="tabpanel" tabindex="0">
          <!---->
        </div>
        <div aria-labelledby="extraTabArtist" id="extraPanelArtist" role="tabpanel" tabindex="0" style="display: none;">
          <!---->
        </div>
        <div aria-labelledby="extraTabAlbum" id="extraPanelAlbum" role="tabpanel" tabindex="0" style="display: none;">
          <!---->
        </div>
        <div aria-labelledby="extraTabAlbum" id="extraPanelAlbum" role="tabpanel" tabindex="0" style="display: none;">
          <!---->
        </div>
      </div>
    </div>
  </section>
  <div class="modal-wrapper">
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
</div>
`;
